ddtrace/appsec/ddwaf/ddwaf_types.py

```diff
@@ -187,16 +186,15 @@ def truncate_string(string):
                     max_depth=max_depth - 1,
                     max_string_length=max_string_length,
                 )
-                if obj.type:  # discards invalid objects
-                    ddwaf_object_map_add(map_o, res_key, obj)
+                ddwaf_object_map_add(map_o, res_key, obj)
         elif struct is not None:
             struct = str(struct)
             if isinstance(struct, bytes):  # Python 2
                 ddwaf_object_string(self, truncate_string(struct))
             else:  # Python 3
                 ddwaf_object_string(self, truncate_string(struct.encode("UTF-8", errors="ignore")))
         else:
-            ddwaf_object_invalid(self)
+            ddwaf_object_null(self)
```

Comment:
probably still worth distinguishing between a `None` value and something unknown

Reply:
At that point, it can only be None, because unknown is handled by the previous case

Reply:
yes, but in this case there are no unknown values: what happens if the string conversion fail?